bundles/org.openhab.binding.volvooncall/src/main/java/org/openhab/binding/volvooncall/internal/handler/VehicleHandler.java

```diff
@@ -159,15 +165,17 @@ private void queryApiAndUpdateChannels() {
                 getThing().getChannels().stream().map(Channel::getUID)
                         .filter(channelUID -> isLinked(channelUID) && !LAST_TRIP_GROUP.equals(channelUID.getGroupId()))
                         .forEach(channelUID -> {
-                            State state = getValue(channelUID.getIdWithoutGroup(), vehicleStatus, vehiclePosition);
+                            State state = getValue(channelUID.getGroupId(), channelUID.getIdWithoutGroup(),
+                                    vehicleStatus, vehiclePosition);
 
                             updateState(channelUID, state);
                         });
                 updateTrips(bridgeHandler);
             } catch (JsonSyntaxException | IOException e) {
                 logger.warn("Exception occurred during execution: {}", e.getMessage(), e);
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
-                refreshJob = null;
+                refreshJob.get().cancel(true);
```

Comment:
This will throw `NoSuchElementException` if the Optional is empty

Reply:
Reverted